dts/bindings/i2c/i2c-device.yaml

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2017, Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: I2C Device Base Structure
+id: i2c-device
+version: 0.1
+
+description: >
+    This binding gives the base structures for all i2c devices
+
+parent-node:
+  - bus: i2c
```

Comment:
I think here we can do a bit more simple here:
parent-node: bus-master

Then in the script, test this single condition:
if parent-node is bus-master:
  *generate defs with <PARENT_BASE_LABEL>_ prefix
  *generate def <PARENT_BASE_LABEL>_I2C_BUS = parent label